internal/services/apimanagement/api_management_resource.go

```diff
@@ -7,32 +7,40 @@ import (
 	"context"
 	"fmt"
 	"log"
+	"net/http"
 	"strconv"
 	"strings"
 	"time"
 
-	"github.com/Azure/azure-sdk-for-go/services/apimanagement/mgmt/2021-08-01/apimanagement" // nolint: staticcheck
+	"github.com/Azure/go-autorest/autorest"
```

Comment:
`hashicorp/go-azure-sdk` no longer uses `Azure/go-autorest` for APIManagement - so we can't rely on casting errors using it - can we just check the Status Codes being returned instead?

Reply:
@tombuildsstuff thanks for your feedback:).